app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\UserRequest;
+use App\Models\User;
+use App\Repositories\User\UserInterface;
+use Illuminate\Http\Request;
+
+class UserController extends Controller
+{
+
+    protected $userRepository;
+
+    public function __construct(UserInterface $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $user = $this->userRepository;
```

Comment:
kết quả trả về của $this->userRepository là gì ạ, có thể chụp ảnh cho em xem được không?

Reply:
ak` a coi code của bên BaseReposi thì nó đã khởi tạo cái model cụ thể là  cái model User trong nớ nên nếu in ra thì nó sẽ là 1 cái đối tượng của cái User, hình a có chụp bên dưới nhé:
![image](https://user-images.githubusercontent.com/51468300/62178104-c63be080-b371-11e9-8417-a483ea336541.png)

Reply:
nếu như này thì gọi tới giá trị của nó như nào ạ :-?
tại sao mình không dùng Repository gọi tới all() thì nhìn kết quả trả về nó sẽ gọn hơn bao nhiêu, đúng không ạ

Reply:
a`k cái này khi mình muốn gọi tới giá trị của nó thì mình làm như thế này vẫn đc nè e: 
$user = $this->userRepository;
$user->phone = '11111111111';

echo $user->phone; #=> 11111111111